cloud-sql/mysql/servlet/src/main/java/com/example/cloudsql/ConnectionPoolContextListener.java

```diff
@@ -0,0 +1,142 @@
+    // See https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory for details.
+    config.addDataSourceProperty("socketFactory", "com.google.cloud.sql.mysql.SocketFactory");
+    config.addDataSourceProperty("cloudSqlInstance", CLOUD_SQL_INSTANCE_NAME);
+    config.addDataSourceProperty("useSSL", "false");
+
+    // ... Specify additional connection properties here.
+
+    // [START_EXCLUDE]
+
+    // [START cloud_sql_max_connections]
+    // maximumPoolSize limits the total number of concurrent connections this pool will keep. Ideal
+    // values for this setting are highly variable on app design, infrastructure, and database.
+    config.setMaximumPoolSize(5);
+    // [END cloud_sql_max_connections]
+
+    // [START cloud_sql_connection_timeout]
+    // setConnectionTimeout is the maximum number of milliseconds to wait for a connection checkout.
+    // Any attempt to retrieve a connection from this pool that exceeds the set limit will throw an
+    // SQLException.
+    config.setConnectionTimeout(10000); // 10 seconds
+    // [END cloud_sql_connection_timeout]
+
+    // [START cloud_sql_connection_backoff]
+    // Hikari automatically delays between failed connection attempts, eventually reaching a
+    // maximum delay of `connectionTimeout / 2` between attempts.
+    // [END cloud_sql_connection_backoff]
+
+    // [START cloud_sql_connection_lifetime]
+    // maxLifetime is the maximum possible lifetime of a connection in the pool. Connections that
+    // live longer than this many milliseconds will be closed and reestablished between uses. This
+    // value should be several minutes shorter than the database's timeout value to avoid unexpected
+    // terminations.
+    config.setMaxLifetime(1800000); // 30 minutes
+    // [END cloud_sql_connection_lifetime]
+
+    // [START cloud_sql_idle_connections]
+    // minimumIdle is the minimum number of idle connections Hikari maintains in the pool.
+    // Additional connections will be established to meet this value unless the pool is full.
+    config.setMinimumIdle(5);
+    // idleTimeout is the maximum amount of time a connection can sit in the pool. Connections that
+    // sit idle for this many milliseconds are retried if minimumIdle is exceeded.
+    config.setIdleTimeout(600000); // 10 minutes
+    // [END cloud_sql_idle_connections]
+    // [END_EXCLUDE]
+
+    // Initialize the connection pool using the configuration object.
+    DataSource pool = new HikariDataSource(config);
+    // [END cloud_sql_mysql_connection_pool]
+    return pool;
+  }
+
+  private void createTableSchema(DataSource pool) {
+    // Safely attempt to create the table schema.
+    try (Connection conn = pool.getConnection()) {
+      PreparedStatement createTableStatement = conn.prepareStatement(
+          "CREATE TABLE IF NOT EXISTS votes ( "
+              + "vote_id SERIAL NOT NULL, time_cast timestamp NOT NULL, canidate CHAR(6) NOT NULL, "
```

Comment:
what does `canidate` mean?

Reply:
`canidate` is the subject of the vote (e.g. `tabs` or `spaces`).

Reply:
is it candidate?

Reply:
Sorry yes - my bad. Fixed.